### ISSUE TITLE
Question: Race conditions

### ISSUE THREAD

**Issue by megakid** [markedup-mobi/file-lock#8]
Hi there,

I was looking at the implementation here and wondering what safety there is over two processes accessing the same lock at exactly the same time and both returning true?  I see the File.Create - will this lock the file (on the file system) and stop another File.Create from succeeding?

Also I am generating a pull request to introduce hostname to the lock file so PID clashes across multiple hosts cannot occur - we are looking to use network storage and this file locking to provide extremely basic active/passive services across multiple hosts.

PS. I also a massive fan of akka.net - looking to do an internal prototype soon that should rule this tactical active-passive approach null-and-void in the long term.

James

**Comment by tamusjroyce** [comment]
Yes. The File.Create is an atomic hardware / OS lock. Meaning only once process can use it at a time. If two try, an exception occurs with the latter one. Thus the race ends with a winner.

But there kind of is an issue with this library. It does not hold the lock open. This class does not implement IDisposable to do that either. I do like the fact you can lock a file past the lifespan of the application. But it is often more practical to just hold it open. And if another program comes along and can write to the lock, it is no longer locked.
